src/e2e-test/java/io/cdap/plugin/bigquerymultitable/stepsdesign/BQMTSink.java

```diff
@@ -179,12 +151,63 @@ public void enterBigQueryMultiTableSinkCmekPropertyAsMacroArgumentIfCmekIsEnable
+  @Then("Enter BiqQueryMultiTable sink property encryption key name {string} if cmek is enabled")
   public void enterBiqQueryMultiTableSinkPropertyEncryptionKeyNameIfCmekIsEnabled(String cmek) throws IOException {
-    String cmekBQ =  PluginPropertyUtils.pluginProp(cmek);
+    String cmekBQ = PluginPropertyUtils.pluginProp(cmek);
     if (cmekBQ != null) {
       BQMTActions.enterBQMTCmekProperty(cmekBQ);
       BeforeActions.scenario.write("Entered encryption key name - " + cmekBQ);
     }
   }
+
+  @Then("Validate records transferred to target BQMT tables {string} is equal to number of records from " +
+    "source BQ table")
+  public void validateRecordsTransferredToTargetBQMTTablesIsEqualToNumberOfRecordsFromSourceBQTable
+    (String targetBQMTTables)
+    throws IOException, InterruptedException {
+    String[] targetBQMTs = PluginPropertyUtils.pluginProp(targetBQMTTables).split(",");
+    int targetTotalRecordsCount = 0;
+    for (String targetBQTable : targetBQMTs) {
+      targetTotalRecordsCount += BigQueryClient.countBqQuery(targetBQTable);
+    }
+    BeforeActions.scenario.write("Number of records transferred to target tables" + targetTotalRecordsCount);
+    int sourceBQRecordsCount = BigQueryClient.countBqQuery(TestSetupHooks.bqSourceTable);
+    BeforeActions.scenario.write("Number of records from source table" + sourceBQRecordsCount);
+    Assert.assertEquals(sourceBQRecordsCount, targetTotalRecordsCount);
+  }
+
+  @Then("Validate the cmek key {string} of target BQMT tables {string} if cmek is enabled")
+  public void validateTheCmekKeyOfTargetBQMTTablesIfCmekIsEnabled(String cmek, String targetBQMTTables)
+    throws IOException {
+    String cmekBQ = PluginPropertyUtils.pluginProp(cmek);
+    if (cmekBQ != null) {
+      String[] targetBQMTs = targetBQMTTables.split(",");
+      for (String targetBQTable : targetBQMTs) {
+        Assert.assertTrue("Cmek key of target BigQuery table " + targetBQTable + " should be equal to " +
+                            "cmek key provided in config file",
+                          BigQueryClient.verifyCmekKey(targetBQTable, cmekBQ));
+      }
+      return;
+    }
+    BeforeActions.scenario.write("CMEK not enabled");
+  }
+
+  @Then("Validate records transferred to target BQMT tables {string}")
+  public void validateRecordsTransferredToTargetBQMTTables(String bqmtTables) throws IOException, InterruptedException {
+    int outCountRecords = 0;
+    String[] targetTableRecordCount = PluginPropertyUtils.pluginProp(bqmtTables).split(",");
+    for (String table : targetTableRecordCount) {
+      int countTableRecords = countOfNoOfRecords(table);
+      outCountRecords = outCountRecords + countTableRecords;
+    }
+    Assert.assertTrue(outCountRecords > 0);
+    Assert.assertEquals(outCountRecords, (long) this.recordIn());
+  }
+
+  int countOfNoOfRecords(String tableName) throws IOException, InterruptedException {
```

Comment:
Remove this. Similar method (countOfNoOfRecordsTransferredToBigQueryIn) available in CDFHelper.

Reply:
Required and changed the name as suggested.